app/policies/document_policy.rb

```diff
@@ -0,0 +1,52 @@
+class DocumentPolicy
+  attr_reader :user, :document_class
+
+  def initialize(user, document_class)
+    @user = user
+    @document_class = document_class
+  end
+
+  def user_organisation_owns_document_type?
+    document_class.organisations.include?(user.organisation_content_id)
+  end
+
+  def departmental_editor
+    user_organisation_owns_document_type? && user.permissions.include?('editor')
+  end
+
+  def writer
+    user_organisation_owns_document_type?
+  end
+
+  def gds_editor
+    user.gds_editor?
+  end
+
+  def index?
+    gds_editor || departmental_editor || writer
+  end
+
+  def new?
+    gds_editor || departmental_editor || writer
```

Comment:
You can condense the repetition by using `alias_method` (see the [signon code](https://github.com/alphagov/signonotron2/blob/master/app/policies/api_user_policy.rb) for an example).